pkg/template/templateprocessing/template.go

```diff
@@ -84,6 +85,10 @@ func (p *Processor) Process(template *templateapi.Template) field.ErrorList {
 		// referenced namespace.
 		stripNamespace(item)
 
+		gvk := item.GetObjectKind().GroupVersionKind()
+		legacygroupification.OAPIToGroupifiedGVK(&gvk)
+		item.GetObjectKind().SetGroupVersionKind(gvk)
```

Comment:
is it possible to do this only when we detect the non-groupified items?

Reply:
> is it possible to do this only when we detect the non-groupified items?

It's a no-op otherwise.  I don't like branches.